gluonnlp/data/batchify.py

```diff
@@ -66,23 +67,38 @@ def _pad_arrs_to_max_length(arrs, pad_axis, pad_val, use_shared_mem=False):
     return ret, original_length
 
 
-def _stack_arrs(arrs, use_shared_mem=False):
+def _stack_arrs(arrs, use_shared_mem, dtype):
     if isinstance(arrs[0], mx.nd.NDArray):
+        dtype = arrs[0].dtype if dtype is None else dtype
         if use_shared_mem:
-            out = mx.nd.empty((len(arrs),) + arrs[0].shape, dtype=arrs[0].dtype,
+            out = mx.nd.empty((len(arrs),) + arrs[0].shape, dtype=dtype,
                               ctx=mx.Context('cpu_shared', 0))
             return mx.nd.stack(*arrs, out=out)
         else:
             return mx.nd.stack(*arrs)
     else:
         out = np.asarray(arrs)
+        dtype = out.dtype if dtype is None else dtype
         if use_shared_mem:
-            return mx.nd.array(out, ctx=mx.Context('cpu_shared', 0), dtype=out.dtype)
+            return mx.nd.array(out, ctx=mx.Context('cpu_shared', 0), dtype=dtype)
         else:
             return mx.nd.array(out, dtype=out.dtype)
 
 
-class Stack(object):
+class Batchify(object):
+    r""""Base class for applying batchify function on input data.
+
+    Parameters
+    ----------
+    dtype : str or numpy.dtype, default None
+        The value type of the output. If it is set to None, the input data type is used.
+    """
+
+    def __init__(self, dtype=None):
+        self._dtype = dtype
```

Comment:
I suggest manually adding the `dtype` argument to `Stack` and `Pad`. This base class looks redundant.

Reply:
This may be cleaner if we add another batchify function in the future?

Reply:
Since it has only one argument, it would be okay to add it manually to other classes. Just a single-line of code. Also, `Tuple` cannot be inherit `Batchify`.

Reply:
Do we really need to have Tuple inherit Batchify? Tuple is only supposed to wrap other batchify functions.

Reply:
Not really. The main concern is that the base class has just one argument and is equivalent to a single-line of code. I think it would be better to keep the interface as simple as possible and avoid adding too many hierarchies.

Reply:
Okay

Reply:
But we have duplicate doc.